pkg/integration/tests/interactive_rebase/drop_with_custom_comment_char.go

```diff
@@ -0,0 +1,38 @@
+package interactive_rebase
+
+import (
+	"github.com/jesseduffield/lazygit/pkg/config"
+	. "github.com/jesseduffield/lazygit/pkg/integration/components"
+)
+
+var DropWithCustomCommentChar = NewIntegrationTest(NewIntegrationTestArgs{
+	Description:  "Drops a commit with the 'core.commentChar' option set to a custom character",
+	ExtraCmdArgs: []string{},
+	Skip:         false,
+	SetupConfig:  func(config *config.AppConfig) {},
+	SetupRepo: func(shell *Shell) {
+		shell.SetConfig("core.commentChar", ";")
+		shell.CreateNCommits(2)
+	},
+	Run: func(t *TestDriver, keys config.KeybindingConfig) {
+		confirmCommitDrop := func() {
+			t.ExpectPopup().Confirmation().
+				Title(Equals("Delete commit")).
+				Content(Equals("Are you sure you want to delete this commit?")).
+				Confirm()
+		}
+
+		t.Views().Commits().Focus().
+			Lines(
+				Contains("commit 02").IsSelected(),
+				Contains("commit 01"),
+			).
+			Press(keys.Universal.Remove).
+			Tap(confirmCommitDrop).
+			// TODO(gus): look into why this is not failing.
+			Lines(
+				Contains("commit 02").IsSelected(),
+				Contains("commit 01"),
+			)
```

Comment:
When writing this test, I saw that this part ended up not failing like I believe it should. Am I misunderstanding how this works?

Reply:
That's an unfortunate consequence of how the test framework works. After dropping the commit, lazygit runs a refresh of the commits list, but it runs in the background and takes a while to complete. The test framework has logic to wait a while and retry the check if it doesn't succeed right away, but in this case the check succeeds because it still matches the old state before the refresh.

@jesseduffield I wonder if there's some way to teach the test infrastructure that any queued-up refresh requests are still pending, and wait until they are all carried out. That should fix this kind of problem, no?

Reply:
Agreed, it would be great if the test framework had a concept of pending actions. It would take some work to support that though (most of the challenge is in tracking that something is pending when we're using goroutines to do work off the UI thread)

Reply:
Naïvely, it sounds simply like a matter of incrementing a counter when posting a refresh, and decrementing it again when the refresh was handled. But I'm probably missing something...

Reply:
That's certainly something we could do when refreshing, and it should be straightforward to implement, but I'm thinking more broadly about all actions that make use of goroutines for doing work off the UI thread

Reply:
Don't let the Perfect be the enemy of the Good... 😉